build.gradle.kts

```diff
@@ -75,10 +76,36 @@ repositories {
   mavenCentral()
 }
 
+detekt {
+  source.setFrom("$projectDir/src/main")
+  buildUponDefaultConfig = true // preconfigure defaults
+  allRules = false // activate all available (even unstable) rules.
+  config.setFrom("$projectDir/detekt.yml") // point to your custom config defining rules to run, overwriting default behavior
+  baseline = file("$projectDir/detekt-baseline.xml") // a way of suppressing issues before introducing detekt
+}
+
+configurations.matching { it.name == "detekt" }.all {
+  resolutionStrategy.eachDependency {
+    if (requested.group == "org.jetbrains.kotlin") {
+      useVersion("1.8.21")
```

Comment:
do you know why this needed? won't it work with 1.8.22?

Reply:
Was running into the behaviour mentioned here in their own documentation and this was their remedy:
[](https://detekt.dev/docs/gettingstarted/gradle/#gradle-runtime-dependencies)